Client/multiplayer_sa/CMultiplayerSA_Postprocess.cpp

```diff
@@ -184,6 +202,24 @@ bool CMultiplayerSA::IsThermalVisionEnabled()
     return (*(BYTE*)0xC402B9 == 1);
 }
 
+std::tuple<bool, float, float> CMultiplayerSA::GetUnderwaterEffect()
+{
+    bool bEnabled = (*(BYTE*)0xC402D3 == 1) || (*(float*)0xC8132C) >= 0.535f;
```

Comment:
You should compare `CWeather::UnderWaterness` with `CPostEffects::m_fWaterFXStartUnderWaterness`, and not the value behind it. Adding names to the addresses also helps others understand what's going on here.

Also, please use `uint8_t` instead of `BYTE`.